src/binfhe/lib/binfhecontext.cpp

```diff
@@ -196,13 +206,61 @@ LWECiphertext BinFHEContext::Encrypt(ConstLWEPrivateKey sk, const LWEPlaintext&
         mod = q;
     LWECiphertext ct = m_LWEscheme->Encrypt(LWEParams, sk, m, p, mod);
 
-    if ((output != FRESH) && (p == 4)) {
-        ct = m_binfhescheme->Bootstrap(m_params, m_BTKey, ct);
+    // BINFHE_OUTPUT is kept as it is for backward compatibility but
+    // this logic is obsolete now and commented out
+    // if ((output != FRESH) && (p == 4)) {
+    //    ct = m_binfhescheme->Bootstrap(m_params, m_BTKey, ct);
+    //}
+
+    return ct;
+}
+
+LWECiphertext BinFHEContext::Encrypt(ConstLWEPublicKey pk, const LWEPlaintext& m, BINFHE_OUTPUT output,
```

Comment:
passing m by const reference may seems as a good decision, but it is needless. LWEPlaintext and LWEPlaintextModulus are integers and should be passed by value. in this case you don't need to make them const and passing by value may be even faster.
on the contrary, "NativeInteger mod" should be passed by const reference as it is an object of a custom class.

Reply:
for now a TODO comment would suffice.